infra/datalad.md

```diff
@@ -0,0 +1,83 @@
+We use Datalad to do provenance tracking for large-scale data processing. 
+
+Penn LINC has their workflow "The Way" documented out [here](https://pennlinc.github.io/docs/TheWay/RunningDataLadPipelines/), and similarly [here] (http://handbook.datalad.org/en/latest/beyond_basics/101-170-dataladrun.html) in the Datalad Handbook. This documentation is meant to provide an extension to modifying this general workflow for your BIDSAPP, data, and cluster, as well as smooth over some problems you might run into. This is adapted from a pyAFQ workflow on Stanford's Sherlock cluster, but can easily be translated to most BIDSAPPS.
+
+Within "The Way", you set up your workflow by running a "bootstrap script" that creates your code and sets up your datalad infrastructure. Typically, starting from one of the Penn LINC scripts is ideal. It might be worth asking Matt and/or Sydney to see if they have any suggestions for which script would be easiest to adapt to your use case, but it depends on how close your container workflow is to another, and idiosyncrasies of your compute cluster. 
+
+### Setup.
+Datalad itself can be installed with `pip install datalad` or `conda install -c conda-forge datalad`. 
+
+To get a boostrap script, you can clone The Way github repo (link). When you run the boostrap script, it will set up the boostrap directory in at the same level as the script, which can be annoying if you're in a Github repo, so you can avoid this by hard link the specific bootstrap script that you want (`ln path/to/target path/to/source`) to a different directory where you want your Datalad work directory to live. As results are created, the directory created by this script can get very large, so OAK or SCRATCH might be a good place.
+
+If you're getting the data from another Datalad workflow, it'll likely be saved as a RIA store and already be a datalad dataset. If not, you'll need to set up your BIDS directory as a [datalad dataset] (https://handbook.datalad.org/en/latest/basics/101-101-create.html). 
+
+If you are using one of the bootstrap scripts right out of the box and have the data stored in a RIA (more on this later), you run it with the following command: 
+
+```bash
+bash bootstrap-script.sh ria+file://[/absolute/path/to/output_ria]\#~data [/path/to/container]
+```
+*Note the leading slashes and lack of trailing slashes. 
+
+So, for example, here is the NKI pyAFQ run with our specific Sherlock pathways. 
+
+```bash
+bash bootstrap-pyAFQ-NKI.sh ria+file:///oak/stanford/groups/jyeatman/NKI_Rockland/QSIPrep/output_ria#~data /scratch/groups/jyeatman/pyafq-container 
```

Comment:
This might have to change quite a bit in the hyak context. And we also think that we could change this to push/pull from/to S3 storage as well. This documentation might be helpful in moving towards that: https://handbook.datalad.org/en/latest/basics/101-139-s3.html